include/seqan3/core/simd/all.hpp

```diff
@@ -6,42 +6,16 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
+ * \brief [DEPRECATED] Meta-header for the simd module.
  * \author Marcel Ehrhardt <marcel.ehrhardt AT fu-berlin.de>
- * \brief Meta-header for the simd module.
+ * \deprecated This header will be removed in SeqAn-3.1.0. Please use seqan3/utility/simd/all.hpp instead.
  */
 
 #pragma once
 
-//!\cond DEV
+#include <seqan3/core/platform.hpp>
 
-/*!\defgroup simd Simd
- * \brief The simd module contains a unified interface to provide simd types and functions used in seqan3.
- * \see https://en.wikipedia.org/wiki/SIMD
- * \ingroup core
- *
- * \todo Make this public again. We made this documentation internal-documentation only for the 3.0.0 release, because
- * the API was not in shape yet. Remove the `cond DEV` and `todo` commands and remove `seqan3::simd` from
- * SEQAN3_DOXYGEN_EXCLUDE_SYMBOLS in `seqan3-doxygen.cmake`.
- *
- * There are different simd implementations (backends), which are auto-selected by seqan3::simd::simd_type_t.
- * \if DEV
- * Namely seqan3::detail::builtin_simd.
- * \endif
- *
- * \todo Describe more what SIMD is and how to use it.
- */
-
-#include <seqan3/core/simd/concept.hpp>
-#include <seqan3/core/simd/debug_stream_simd.hpp>
-#include <seqan3/core/simd/simd_algorithm.hpp>
-#include <seqan3/core/simd/simd_traits.hpp>
-#include <seqan3/core/simd/simd.hpp>
-#include <seqan3/core/simd/view_iota_simd.hpp>
-#include <seqan3/core/simd/view_to_simd.hpp>
+SEQAN3_DEPRECATED_HEADER(
+   "This header is deprecated and will be removed in SeqAn-3.1.0; Please #include <seqan3/utility/simd/all.hpp> instead.")
 
-/*!\namespace seqan3::simd
- * \brief The SeqAn namespace for simd data types, algorithms and meta functions.
- * \sa https://en.wikipedia.org/wiki/SIMD What is SIMD conceptually?
- * \sa https://en.wikipedia.org/wiki/stream_REMOVEMEing_SIMD_Extensions Which SIMD architectures exist?
- */
-//!\endcond
+#include <seqan3/utility/simd/all.hpp>
```

Comment:
I thought there might be an issue if I include `#include <seqan3/utility/simd/all.hpp>` first, that first some other deprecation warning appears that might mislead that you have to use `seqan3/utility/simd/all.hpp` instead of `seqan3/code/simd/all.hpp`. But on second thought this should not be possible, because of our tests.